dayrepo/snippets/views.py

```diff
@@ -225,117 +226,182 @@ def get(self, request, snippet_id):
+                load_mileage=req.getlist("load_mileage")[i],
+                hollow_mileage=req.getlist("hollow_mileage")[i],
+                is_load_situation=req.getlist("is_load_situation")[i],
             )
-            if snippet.is_today_trouble == "on":
-                snippet.is_today_trouble = True
+            if process.is_load_situation == "on":
+                process.is_load_situation = True
             else:
-                snippet.is_today_trouble = False
-            snippet.save()
-
-            # 業務トラブルフォーム保存
-            duties_trouble = DutiesTrouble(
-                id=post_trouble.id,
-                snippet_id=post_trouble.snippet_id,
-                trouble_situation=req.get("trouble_situation"),
-                trouble_cause=req.get("trouble_cause"),
-                trouble_support=req.get("trouble_support"),
-            )
-            post_trouble.delete()
-            duties_trouble.save()
-            
-            
-            # 工程テーブルフォーム保存
-            form_process_count = len(req.getlist("via_point"))
-            for i in range(form_process_count):
-                process = Process(
-                    id=post_process[i].id,
-                    snippet_id=post_process[i].snippet_id,
-                    start_time=req.getlist("start_time")[i + 1],
-                    end_time=req.getlist("end_time")[i + 1],
-                    start_point=req.getlist("start_point")[i + 1],
-                    end_point=req.getlist("end_point")[i + 1],
-                    via_point=req.getlist("via_point")[i],
-                    client=req.getlist("client")[i],
-                    goods=req.getlist("goods")[i],
-                    load_situation=req.getlist("load_situation")[i],
-                    load_mileage=req.getlist("load_mileage")[i],
-                    hollow_mileage=req.getlist("hollow_mileage")[i],
-                    is_load_situation=req.getlist("is_load_situation")[i],
-                )
-                if process.is_load_situation == "on":
-                    process.is_load_situation = True
-                else:
-                    process.is_load_situation = False
-                post_process[i].delete()
-                process.save()
-            post_snippet=snippet
-            # トップ画面へ
-            return redirect(to="snippet_list")
+                process.is_load_situation = False
+            post_process[i].delete()
+            process.save()
+        post_snippet = snippet
+        # トップ画面へ
+        return redirect(to="snippet_list")
+
 snippet_edit = SnippetEditView.as_view()
 
+
+class DbDeletePattern():
+    def snippet_del(self,target_id:int):
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        process = get_list_or_404(
+            Process, snippet_id=target_id
+        )
+        for process_object in process:
+            process_object.delete()
```

Comment:
以前 all_del のほうで指摘した内容です。
こう言うことがあるので、こういった完全に同じ共通処理は共通化したいんですよね〜🤔
```suggestion
        process_list = get_list_or_404(
            Process, snippet_id=target_id
        )
        for process in process_list:
            process.delete()
```